apps/server/src/infra/authorization-client/authorization-client.adapter.ts

```diff
@@ -0,0 +1,67 @@
+import { Inject, Injectable, UnauthorizedException } from '@nestjs/common';
+import { REQUEST } from '@nestjs/core';
+import { RawAxiosRequestConfig } from 'axios';
+import cookie from 'cookie';
+import { Request } from 'express';
+import { ExtractJwt, JwtFromRequestFunction } from 'passport-jwt';
+import { AuthorizationApi, AuthorizationBodyParams } from './authorization-api-client';
+import { AuthorizationErrorLoggableException, AuthorizationForbiddenLoggableException } from './error';
+
+@Injectable()
+export class AuthorizationClientAdapter {
+	constructor(private readonly authorizationApi: AuthorizationApi, @Inject(REQUEST) private request: Request) {}
+
+	public async checkPermissionsByReference(params: AuthorizationBodyParams): Promise<void> {
+		const hasPermission = await this.hasPermissionsByReference(params);
+		if (!hasPermission) {
+			throw new AuthorizationForbiddenLoggableException(params);
+		}
+	}
+
+	public async hasPermissionsByReference(params: AuthorizationBodyParams): Promise<boolean> {
+		const options = this.createOptionParams();
+
+		try {
+			const response = await this.authorizationApi.authorizationReferenceControllerAuthorizeByReference(
+				params,
+				options
+			);
+
+			const hasPermission = response.data.isAuthorized;
+
+			return hasPermission;
+		} catch (error) {
+			throw new AuthorizationErrorLoggableException(error as Error, params);
```

Comment:
error ist vom type unkown ohne instanceof check finde ich den cast unglücklich.

Reply:
Fixed by extending the error handling inside of `AuthorizationErrorLoggableException`!